ext/mysqlnd/mysqlnd_charset.c

```diff
@@ -831,32 +816,32 @@ PHPAPI zend_ulong mysqlnd_cset_escape_slashes(const MYSQLND_CHARSET * const cset
 	const char 	*newstr_s = newstr;
 	const char 	*newstr_e = newstr + 2 * escapestr_len;
 	const char 	*end = escapestr + escapestr_len;
-	bool	escape_overflow = FALSE;
 
 	DBG_ENTER("mysqlnd_cset_escape_slashes");
 	DBG_INF_FMT("charset=%s", cset->name);
 
 	for (;escapestr < end; escapestr++) {
 		char esc = '\0';
-		unsigned int len = 0;
 
-		/* check unicode characters */
-		if (cset->char_maxlen > 1 && (len = cset->mb_valid(escapestr, end))) {
-			/* check possible overflow */
-			if ((newstr + len) > newstr_e) {
-				escape_overflow = TRUE;
-				break;
-			}
-			/* copy mb char without escaping it */
-			while (len--) {
-				*newstr++ = *escapestr++;
+		/* check unicode characters
+		 * Encodings that have a minimum length of 1 are compatible with ASCII.
+		 * So we can skip (for performance reasons) the check to mb_valid for them. */
+		if (cset->char_maxlen > 1 && (*((zend_uchar *) escapestr) > 0x80 || cset->char_minlen > 1)) {
+			unsigned int len = cset->mb_valid(escapestr, end);
```

Comment:
It looked like it could be skipped in the same way in `mysqlnd_cset_escape_quotes`, but isn't there much of a change?

Reply:
You are right, I factored out the check so I could reuse it for this function.